src/Tempest/Database/QueryStatements/AlterTableStatement.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\QueryStatements;
+
+use Tempest\Database\DatabaseDialect;
+use Tempest\Database\QueryStatement;
+
+final class AlterTableStatement implements QueryStatement
+{
+    public function __construct(
+        private readonly string $tableName,
+        private array $statements = [],
+    ) {
+    }
+
+    public function add(QueryStatement $statement): self
+    {
+        $this->statements[] = new AlterActionStatement(AlterStatement::ADD, $statement);
+
+        return $this;
+    }
+
+    public function update(QueryStatement $statement): self
+    {
+        $this->statements[] = new AlterActionStatement(AlterStatement::UPDATE, $statement);
+
+        return $this;
+    }
+
+    public function delete(QueryStatement $statement): self
```

Comment:
Shouldn't the input here simply be `string $name` instead of another statement? For `add` I understand we need it for adding columns, but I don't think it's needed for deleting them?